vignettes/ConvertingPharmacoSetToGDR.Rmd

```diff
@@ -0,0 +1,109 @@
+---
+title: "Converting PharmacoSet Drug Response Data into gDR object" 
+author:
+- name: Jermiah Joseph
+  email: jermiah.joseph@uhn.ca
+- name: Bartosz Czech
+  email: bartosz.w.czech@gmail.com
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{ConvertingPharmacoSetDrugResponseDataintogDRobject}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(PharmacoGx)
+library(gDR)
+library(gDRcore)
+library(gDRimport)
+```
+
+
+# Overview
+The `gDRimport` package is a part of the gDR suite. It helps to prepare raw drug response data for downstream processing. 
+It mainly contains helper functions for importing/loading/validating dose response data provided in different file formats. 
+In collaboration with the BHKLab, `gDRimport` also provides functions that can convert a `PharmacoGx::PharamcoSet` object into a gDR object. 
+With this functionality, users familiar with the gDR suite of packages and methods can utilize the publically available, curated datasets from the PharmacoGx database. 
+The main step in this process is to extract the drug dose-response data from the PharmacoSets and transform them into a `data.frame` that can be used as input for the `gDRcore::runDrugResponseProcessingPipeline`.
+
+
+# Loading a PharmacoSet (PSet)
+Whereas a user might already have a pharmacoset loaded in their R session, if they wish to obtain a different pharmacoset or use the same script in the future we provide a helper function to do so. 
+It helps to have a user directory in which to store all pharmacosets, and by passing this directory into the function as a parameter, the function will also check to see if the PSet exists in the user-definded directory.
+This is to ensure that the PSet is not being re-downloaded if it already has.  
+```{r eval = FALSE}
+pset <- gDRimport:::.getPSet("Tavor_2020")
+pset
+```
+
+```{r, include = FALSE}
+read_mocked_PSets <- function(canonical = FALSE) {
+  readRDS(
+    system.file("extdata", "data_for_unittests", "PSets.rds", package = "gDRimport")
+  )
+}
+pset <- testthat:::with_mock(
+    `PharmacoGx::availablePSets` = read_mocked_PSets,
+    suppressMessages(getPSet(
+      "Tavor_2020", psetDir = system.file("extdata/pset", package = "gDRimport")
+    ))
+  )
+pset
+```
+
+# Converting PharmacoSet to Dataframe for gDR pipeline
+`PharamcoSets` hold data pertaining to the celllines (@sample slot), drugs (@treatment slot), and dose response experiments (@treatmentResponse slot). 
```

Comment:
typo(?): celllines, should be cell lines

Reply:
Done